django/contrib/admin/options.py

```diff
@@ -308,6 +308,21 @@ def get_readonly_fields(self, request, obj=None):
         """
         return self.readonly_fields
 
+    def get_invariable_fields(self, request, obj=None):
+        """
+        Get fields that are either readonly or han't change/add permissions
+        """
+        if hasattr(request, 'user') and (self.has_change_permission(request, obj) or self.has_add_permission(request)):
```

Comment:
please limit line length to 119 chars so horizontal scrolling isn't required to view the patch.